pandas/tests/series/methods/test_to_period.py

```diff
@@ -45,3 +46,10 @@ def test_to_period(self):
         expected = df.copy()
         expected.columns = exp_idx
         tm.assert_frame_equal(df.to_period(axis=1), expected)
+
+    def test_to_period_raises(self, indices):
+        # invalid type , #34067 test
+        idx = indices
+        msg = "unsupported Type RangeIndex"
+        with pytest.raises(TypeError, match=msg):
+            Series([idx]).to_period()
```

Comment:
we want the series index to be parameterised.

```suggestion
    def test_to_period_raises(self, indices):
        # https://github.com/pandas-dev/pandas/issues/33327
        index = indices
        ser = Series(index=index, dtype=object)
        if not isinstance(index, DatetimeIndex):
            msg = f"unsupported Type {type(index).__name__}"
            with pytest.raises(TypeError, match=msg):
                ser.to_period()
```